CodePushPackage.m

```diff
@@ -179,21 +207,122 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
+                    if (error) {
+                        failCallback(error);
+                        return;
+                    }
+                    
+                    [CodePushPackage copyEntriesInFolder:currentPackageFolderPath
+                                              destFolder:newPackageFolderPath
+                                                   error:&error];
+                    if (error) {
+                        failCallback(error);
+                        return;
+                    }
+                    
+                    // Delete files mentioned in the manifest.
+                    NSString *manifestContent = [NSString stringWithContentsOfFile:diffManifestFilePath
+                                                                          encoding:NSUTF8StringEncoding
+                                                                             error:&error];
+                    if (error) {
+                        failCallback(error);
+                        return;
+                    }
+                    
+                    NSData *data = [manifestContent dataUsingEncoding:NSUTF8StringEncoding];
+                    NSDictionary* manifestJSON = [NSJSONSerialization JSONObjectWithData:data
+                                                                                 options:kNilOptions
+                                                                                   error:&error];
+                    NSArray *deletedFiles = manifestJSON[@"deletedFiles"];
+                    for (NSString *deletedFileName in deletedFiles) {
+                        [[NSFileManager defaultManager]
+                         removeItemAtPath:[newPackageFolderPath
+                                           stringByAppendingPathComponent:deletedFileName]
+                         error:&nonFailingError];
+                        
+                        if (nonFailingError) {
+                            NSLog(@"Error deleting file from current package: %@", nonFailingError);
+                            nonFailingError = nil;
+                        }
+                    }
+                }
+                
+                [CodePushPackage copyEntriesInFolder:unzippedFolderPath
+                                          destFolder:newPackageFolderPath
+                                               error:&error];
+                [[NSFileManager defaultManager] removeItemAtPath:unzippedFolderPath
+                                                           error:&nonFailingError];
+                if (nonFailingError) {
+                    NSLog(@"Error deleting downloaded file: %@", nonFailingError);
+                    nonFailingError = nil;
+                }
+                
+                NSString *relativeBundlePath = [self findMainBundleInFolder:newPackageFolderPath
+                                                                   error:&error];
```

Comment:
Formatting is erratic, sometimes aligned on \:, sometimes on aligned on method name, sometimes on nothing.